### ISSUE TITLE
about working with TPUART

### ISSUE THREAD

**Issue by leoujz** [mistay/stknx#2]
read bytes from STKNX, then is there any way to communicate as a TPUART interface? or is it complicated?

**Comment by mistay** [comment]
reading from TPUART is quite easy: all bytes can be read very similar as they would be read from UART.
reading from STKNX is more complicated: all bits have to be decoded regarding to KNX specifications.
this repository focuses on reading from STKNX, as STKNX is cheaper and more sophisticated to read/write from/to.

**Comment by leoujz** [comment]
Thanks. is writing to stknx okay? it seems that code has not been changed for a long time, is there any plan in future?

**Comment by musti22160** [comment]
345 / 5000
Çeviri sonuçları
I have a question regarding the ST-KNX circuit diagram.

I installed the scheme, but when sending data, there is a "click-click" sound on the circuit. If the detection is not wrong, it comes from the ST-KNX integrated. The sound is increased when I remove the R68 resistor, while the sound decreases significantly with the R68 attached.
Have you ever encountered a situation like this?
What is your solution?

**Comment by mistay** [comment]
> 345 / 5000
> Çeviri sonuçları
> I have a question regarding the ST-KNX circuit diagram.
> 
> I installed the scheme, but when sending data, there is a "click-click" sound on the circuit. If the detection is not wrong, it comes from the ST-KNX integrated. The sound is increased when I remove the R68 resistor, while the sound decreases significantly with the R68 attached.
> Have you ever encountered a situation like this?
> What is your solution?

no, sorry. never ever encountered a situation like this. no click-souns at all.

**Comment by musti22160** [comment]
> > 345 / 5000
> > Çeviri sonuçları
> > I have a question regarding the ST-KNX circuit diagram.
> > I installed the scheme, but when sending data, there is a "click-click" sound on the circuit. If the detection is not wrong, it comes from the ST-KNX integrated. The sound is increased when I remove the R68 resistor, while the sound decreases significantly with the R68 attached.
> > Have you ever encountered a situation like this?
> > What is your solution?
> 
> no, sorry. never ever encountered a situation like this. no click-souns at all.

Thank you, I want to solve this problem. It seems to me that it will fail over time

**Comment by musti22160** [comment]
dear mistay,

If the pcb is still in your hand and you are running it, the sound I am talking about can be heard with the human ear.
Can you check by taking the sound recording of the processor with your mobile phone. 3 out of 5 people in our team said they could not hear this sound. After recording the sound, everyone heard this sound.

We continue to search for a solution. Because we think that the integration is difficult and will cause problems in the long run.

**Comment by mistay** [comment]
> dear mistay,
> 
> If the pcb is still in your hand and you are running it, the sound I am talking about can be heard with the human ear.
> Can you check by taking the sound recording of the processor with your mobile phone. 3 out of 5 people in our team said they could not hear this sound. After recording the sound, everyone heard this sound.
> 
> We continue to search for a solution. Because we think that the integration is difficult and will cause problems in the long run.

please give me some days to achieve this ...

**Comment by musti22160** [comment]
Thank you very much for your interest. I will wait for your reply.

**Comment by mistay** [comment]
> Thank you very much for your interest. I will wait for your reply.

may i ask for what reason you'd like to fix that issue? will you put it into a commercial product? love to hear what your plans are ...

**Comment by musti22160** [comment]
Yes, I want to use it in a commercial product. That's why I don't want it to cause problems in the long run.

**Comment by mistay** [comment]
cool, are is there any public info about that upcoming product. i'm just curious.

**Comment by musti22160** [comment]
Being open to everyone is a bit difficult for commercial reasons. But if you can give your mail or skype address, I can share a few tricks on hardware.
